app/models/git_repository.rb

```diff
@@ -47,7 +47,7 @@ def clone!(from: repository_url, to: repo_cache_dir, mirror: false)
 
   def commit_from_ref(git_reference, length: 7)
     return unless ensure_local_cache!
-    command = ['git', 'describe', '--long', '--tags', '--all', "--abbrev=#{length || 40}", git_reference]
+    command = ['git', 'describe', '--always', '--long', '--tags', '--all', "--abbrev=#{length || 40}", git_reference]
```

Comment:
This method seems to try to obtain the full commit hash. Why don't we use `git rev-parse`?